src/pages/studyView/charts/barChart/BarChart.tsx

```diff
@@ -74,7 +84,37 @@ export default class BarChart extends React.Component<IBarChartProps, {}>
 
     constructor(props: IBarChartProps) {
         super(props);
+        this.state = {
```

Comment:
we mainly use mobx for state management so instead of this.state and setState we use decorators like @observable and @action. can we look into this and see if we can do that?